### DISCUSSION TITLE
Welcome to ClassicMQTT Discussions!

### DISCUSSION THREAD

**Discussion by graham22** [ClassicDIY/ClassicMQTT#23] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by mcsarge** [comment]
**Cool**

**Comment by jamien82** [comment]
Hi Graham,
Is there a way to combine both your serial adapter and your mqtt code together to plug in and transmit over mqtt wifi. 

Also are you able to use separated values in mqtt to more easily use the information?

You have really come a long way with this project since I suggested you add a consumption field
Thanks
Jamie

**Comment by graham22** [comment]
I assume you're referring to the esp implementation, you want to have it connect to a the modbus serial RTU interface and emit MQTT?

You could clone the project and do what you want with it but if I change this one then the android app would also have to change.

**Comment by qcda1** [comment]
Hello Graham. I've been using classic.modbusdecoder.py in my off-grid monitoring system and I greatly appreciate it. With the latest version of pymodbus (3.8.2), I realized there was an issue when inquiring registers of multiple classics. All returned values were the ones from the first classic. I saw that when closing modbusClient, it had to be initialized to None (modbusClient = None) for the code to reopen a connection with the other classics.

Also, when I upgraded to latest version of pymodbus, I realized they deprecated BinaryPayloadDecoder in pymodbus.payload. So I isolated it in one Payload.py file with all it needed to preserve the functionality of BinaryPayloadDecoder. Since it deals with Modicon PLC memory map that I believe will never change, I'm quite confident the new Payload.py file will work as long as the word boundary structure of the Modicon memory map doesn't change in the classics.

I will send to this discussion (unless there is a better way. I'm new to Github) the Payload.py file and classic.modbusdecoder.py shortly for your information.

Regards,

Daniel.

**Comment by graham22** [comment]
Thanks Daniel, [ Mathew Sargent ](https://github.com/mcsarge) was the author for the python version of ClassicMQTT ( I did the ESP32 version), just want to keep him in the loop.
You can create a PR with your update or send me the files and I'll add them...

thanks
Graham
